api/src/utils/helpers.ts

```diff
@@ -1,6 +1,53 @@
-import { IPitch } from 'ssw-common';
+import { IPitch, IUser } from 'ssw-common';
+import { PitchSchema } from '../models/pitch';
+import { UserSchema } from '../models/user';
 
 export const isPitchClaimed = (pitch: IPitch): boolean =>
   Object.entries(pitch.teams).every(
     (team) => team[1].current === team[1].target,
   );
+
+export const santitizePitch = (pitch: PitchSchema): IPitch => ({
```

Comment:
I'm not entirely sure I understand what these sanitize methods are doing. Is it to coerce the schema types returned from the DB to the interfaces? If so, would a cast do the same thing?

Reply:
TypeScript is only a annotation based language to guide the user along what types things are. I can cast it to a IPitch yes but doesn't remove the data. See [here](https://stackoverflow.com/questions/50378612/how-to-cast-object-to-another-type-and-remove-unneeded-fields-in-typescript).

> There is no casting in TypeScript because that's not how TypeScript works. TS is a type layer on top of JS. When you assign a type in TS you don't set a type, you annotate that the given expression is of a certain type. When the TS code is transpiled to JS, all type information is stripped.
>
> Or in other words: the type system is JS, with TS you just announce that a variable is of a certain type. More often than never it happens that you assign the wrong type at design time and get suprised during debug that the variable has a completely different type than expected.
>
> If you want to be sure that a property is removed from an object, you need to go the JS way (and - if necessary - annotate the result with TS). Check this Q&A to see how to remove properties from an object.

So the sanitize function removes all of the Schema stuff that was getting thrown in there.